Signal/src/ViewControllers/ConversationView/Cells/OWSGenericAttachmentView.m

```diff
@@ -116,25 +117,37 @@ - (void)createContents
     [contentView autoPinEdgeToSuperviewEdge:ALEdgeTop withInset:self.vMargin];
     [contentView autoPinEdgeToSuperviewEdge:ALEdgeBottom withInset:self.vMargin];
 
+    OWSBezierPathView *iconCircleView = [OWSBezierPathView new];
+    UIColor *iconColor
+        = (self.isIncoming ? [UIColor colorWithRGBHex:0x9e9e9e] : [self foregroundColorWithOpacity:0.15f]);
+    iconCircleView.configureShapeLayerBlock = ^(CAShapeLayer *_Nonnull layer, CGRect bounds) {
+        CGFloat radius = MIN(bounds.size.width, bounds.size.height) * 0.5f;
+        layer.path = [UIBezierPath bezierPathWithRoundedRect:bounds cornerRadius:radius].CGPath;
+        layer.fillColor = iconColor.CGColor;
+    };
+    [contentView addSubview:iconCircleView];
+    [iconCircleView autoPinEdgeToSuperviewEdge:ALEdgeLeft withInset:self.iconHMargin];
+    [iconCircleView autoVCenterInSuperview];
+    [iconCircleView autoSetDimension:ALDimensionWidth toSize:self.iconSize];
+    [iconCircleView autoSetDimension:ALDimensionHeight toSize:self.iconSize];
+    [iconCircleView setContentHuggingHigh];
+
     UIImage *image = [UIImage imageNamed:@"generic-attachment-small"];
     OWSAssert(image);
     UIImageView *imageView = [UIImageView new];
     imageView.image = [image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
     imageView.tintColor = self.bubbleBackgroundColor;
-    imageView.backgroundColor
-        = (self.isIncoming ? [UIColor colorWithRGBHex:0x9e9e9e] : [self foregroundColorWithOpacity:0.15f]);
-    imageView.layer.cornerRadius = MIN(imageView.bounds.size.width, imageView.bounds.size.height) * 0.5f;
     [contentView addSubview:imageView];
-    [imageView autoPinEdgeToSuperviewEdge:ALEdgeLeft withInset:self.iconHMargin];
-    [imageView autoVCenterInSuperview];
-    [imageView autoSetDimension:ALDimensionWidth toSize:self.iconSize];
-    [imageView autoSetDimension:ALDimensionHeight toSize:self.iconSize];
+    [imageView autoPinEdge:ALEdgeLeft toEdge:ALEdgeLeft ofView:iconCircleView];
```

Comment:
TIOLI: it seems like the idiomatic thing to do here is make imageView a subview of iconCircleView with 

```
[iconCircleView addSubview:imageView];
[imageView autoPinEdgesToSuperviewEdges];
```

Reply:
Unfortunately that doesn't work.  TIL `OWSBezierPathView` won't render subviews.  I'm not sure why... could it be because we're constantly recreates the `OWSBezierPathView`'s layers?

Reply:
Ah right... probably because we do:

```
    for (CALayer *layer in self.layer.sublayers) {
        [layer removeFromSuperlayer];
    }
```

Probably not worth fixing for this use.